ui.apps/src/main/content/jcr_root/apps/venia/clientlibs/common/js/CommerceApi.js

```diff
@@ -0,0 +1,203 @@
+                throw new Error('The commerce API is not properly initialized. The "endpoint" property is missing from the initialization object');
+            }
+
+            this.rootEndpoint = props.endpoint;
+        }
+
+        /**
+         * Issues a request to the supplied URL using the provided parameters
+         * @param url
+         * @param params
+         * @returns {Promise<any>} the JSON response or throws an error
+         * @private
+         */
+        async _fetch(url, params) {
+            let response = await fetch(url, params);
+            if (!response.ok) {
+                let message = await response.text();
+                throw new Error(message);
+            }
+            return response.json();
+        }
+
+        /**
+         * Performs an update operation (POST or PUT).
+         * @param url the request URL
+         * @param params the URL parameters
+         * @param method the method to use for the update - POST or PUT
+         * @returns {Promise<any>}
+         * @private
+         */
+        async _update(endpoint, params, method) {
+            let url = `${this.rootEndpoint}${endpoint}`;
+            let defaultParams = {
+                method,
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+            };
+
+            let extendedParams = Object.assign({}, params, defaultParams);
+
+            return this._fetch(url, extendedParams);
+        }
+
+        async _post(endpoint, params) {
+            return this._update(endpoint, params, 'POST');
+        }
+
+        async _put(endpoint, params) {
+            return this._update(endpoint, params, 'PUT');
+        }
+
+        async _get(endpoint) {
+            let url = `${this.rootEndpoint}${endpoint}`;
+            return this._fetch(url);
+        }
+
+        async _delete(endpoint) {
+            let url = `${this.rootEndpoint}${endpoint}`;
+            let params = {"method": "DELETE"};
+
+            return this._fetch(url, params);
+        }
+
+        /**
+         * Retrieves the cart data in JSON format
+         * @param id the cart id
+         * @returns {Promise<T>}
+         */
+        async getCart(id) {
+            return await this._get(endpoints.guestcarts.byId(id));
+        }
+
+        /**
+         * Creates an empty shopping cart.
+         * @returns {Promise<*>}
+         */
+        async createCart() {
+            return await this._post(endpoints.guestcarts.create);
+        }
+
+        /**
+         * Updates a cart entry
+         * @param cartId
+         * @param itemId
+         * @param sku
+         * @param qty
+         * @param quoteId
+         * @returns {Promise<*>}
+         */
+        async updateCartEntry(cartId, itemId, {sku, qty, quoteId}) {
+            let url = `${endpoints.guestcarts.itemOperation(cartId, itemId)}`;
+            let params = {
+                cartItem: {
+                    sku,
```

Comment:
Silly question but ... shouldn't we have object keys for `sku` and `qty`? Same question for `postCartEntry`

Reply:
In ES6 { sku } is equivalent to { sku : sku }